model/inline/src/main/scala/aqua/model/inline/tag/ForTagInliner.scala

```diff
@@ -41,14 +42,30 @@ final case class ForTagInliner(
           s"non-box type variable '$iterable' in 'for' expression."
         )
     }
-    _ <- Exports[S].resolved(item, VarModel(n, elementType))
+    itemVar = VarModel(n, elementType)
+    _ <- Exports[S].resolved(item, itemVar)
+    pref <- keyValue match {
+      case None => State.pure(None)
+      case Some(kv) =>
+        for {
+          keyName <- Mangler[S].findAndForbidName(kv.key)
+          _ <- Exports[S].resolved(kv.key, VarModel(keyName, ScalarType.string))
+          valueName <- Mangler[S].findAndForbidName(kv.value)
+          _ <- Exports[S].resolved(kv.value, VarModel(valueName, elementType))
+        } yield {
+          Some(SeqModel.wrap(
+            FlattenModel(itemVar.withProperty(IntoFieldModel("key", ScalarType.string)), keyName).leaf,
+            FlattenModel(itemVar.withProperty(IntoFieldModel("value", elementType)), valueName).leaf
```

Comment:
Is it possible to not always flatten those values, but export them with properties?
So that whoever uses this value will flatten it if it is needed?